geopandas/tests/test_plotting.py

```diff
@@ -871,6 +872,47 @@ def test_plot(self):
         self.df.plot()
 
 
+class TestColorParamArray:
```

Comment:
There is a TestNonuniformGeometryPlotting class below for tests involving mixture of points/lines/polygons, so this might fit there?

Reply:
I think it's really the same.  the tests I have constructed need to be in place  and need geometry that is constructed to ensure permutations of geometry types and multi-geometry types work.  Hence refactor really would be shift code out of new test class into existing class or leave in it's own